File: app/src/main/java/com/pr0gramm/app/feed/Vote.java
package com.pr0gramm.app.feed;

/**
 */
public enum Vote {
    DOWN(-1), NEUTRAL(0), UP(1);
    private final int voteValue;

    Vote(int voteValue) {
        this.voteValue = voteValue;
    }

    public int getVoteValue() {
        return voteValue;
    }
}

File: app/src/main/java/com/pr0gramm/app/orm/Vote.kt
package com.pr0gramm.app.orm

/**
 */
enum class Vote(val voteValue: Int) {
    DOWN(-1), NEUTRAL(0), UP(1);

    companion object {
        fun ofVoteValue(value: Int): Vote {
            return when {
                value > 0 -> UP
                value < 0 -> DOWN
                else -> NEUTRAL
            }
        }
    }
}
